### PR TITLE
OCPBUGS-3822: improve subscription exists alert message

### PR DESCRIPTION
**Pull request by rhamilto**
Updated alert:
Screenshot 2023-01-20 at 4 08 08 PM

In order to avoid the following error when deleting the existing Subscription, we need to first [check if the CSV exists](https://github.com/openshift/console/pull/12454/files#diff-5e377323073b833a3b2cb0aa8937636dc4aaf36ab1d136bff79ffd92ea2c83bcR151) before trying to delete it in the event it was already deleted:
Screenshot 2023-01-20 at 3 54 06 PM

### CODE REVIEW

**Comment by rhamilto** [comment]
@tlwu2013, thoughts on the error in the `Uninstall Operator?` modal when deleting the existing Subscription since the CSV was already deleted?  See https://issues.redhat.com/browse/OCPBUGS-3822 for full details.

**Comment by rhamilto** [comment]
/hold

broke test

**Comment by rhamilto** [comment]
/hold cancel

test fixed

**Comment by rhamilto** [comment]
@TheRealJon, mind reviewing?

**Comment by rhamilto** [comment]
@TheRealJon or @jhadvig, mind reviewing so we can get this bug closed?

**Review by jhadvig — APPROVED** [review]
/lgtm
/approve

File: frontend/packages/operator-lifecycle-manager/src/components/modals/uninstall-operator-modal.tsx [hunk 1]
```diff
@@ -125,14 +126,29 @@ export const UninstallOperatorModal: React.FC<UninstallOperatorModalProps> = ({
...
+          name: subscription.status.installedCSV,
+          ns: subscription.metadata.namespace,
+        });
+        return true;
+      } catch (err) {
+        // eslint-disable-next-line no-console
+        console.log(err.message);
+        return false;
+      }
+    };
```
Comment (TheRealJon) [inline frontend/packages/operator-lifecycle-manager/src/components/modals/uninstall-operator-modal.tsx:147]: What happens when the request fails with something other than a 404?

**Review by TheRealJon — COMMENTED** [review]
I don't think we can assume that a failed request for the installed CSV means it doesn't exist. Do we need a different message for the case where we can't determine the existence of the CSV due to a failed response other than a 404?

File: frontend/packages/operator-lifecycle-manager/src/components/modals/uninstall-operator-modal.tsx (hunk 1 above)
Comment (rhamilto) [reply frontend/packages/operator-lifecycle-manager/src/components/modals/uninstall-operator-modal.tsx:147]: Good point.  We need to report out those other errors.

/hold

File: frontend/packages/operator-lifecycle-manager/src/components/modals/uninstall-operator-modal.tsx [hunk 2]
```diff
@@ -125,14 +129,30 @@ export const UninstallOperatorModal: React.FC<UninstallOperatorModalProps> = ({
...
+    const clusterServiceVersionExists = async () => {
+      try {
+        await k8sGetResource({
+          model: ClusterServiceVersionModel,
+          name: subscription.status.installedCSV,
+          ns: subscription.metadata.namespace,
+        });
+        return true;
+      } catch (err) {
+        if (err.json.code !== 404) {
```
Comment (rhamilto) [inline frontend/packages/operator-lifecycle-manager/src/components/modals/uninstall-operator-modal.tsx:146]: @TheRealJon, updated.  PTAL.

**Review by TheRealJon — APPROVED** [review]
/lgtm

**Pull request merged**